VueJS/index.js

```diff
@@ -4,10 +4,47 @@ import ListItem from './components/ListItem.js'
 new Vue({
   el: '#app',
   data: {
+    newTodoText: '',
+    counter:0,
+    editmode:[
+      false,false,false
+      ],
+    tempValues:[
+      { id: '0', message: null, },
+      { id: '1', message: null, },
+      { id: '2', message: null, }
+    ],
     todos: [
-      { id: '1', message: 'Foo', },
-      { id: '2', message: 'Bar', },
-      { id: '3', message: 'Baz', }
+      { id: '0', message: 'Foo', },
+      { id: '1', message: 'Bar', },
+      { id: '2', message: 'Baz', }
     ] // feel free to put the `todos` array where you find it most suitable
+    ,nextTodoId: 3
+  },
+  methods:{
+    enableEditing: function(index){
+      this.tempValues[index].message= this.todos[index].message
+      this.$set(this.editmode, index, true)
```

Comment:
This is again not very good modeling/abstraction. Why do we have the `index` here?

Could you structure your code into components with a dedicated responsibility? If you had an `Item` component, your component would receive the click event and conditionally render the input field.